In [5]:
from kaggle_environments import make

env = make("kaggriculture", debug=True)

In [2]:
from sb3_contrib import MaskablePPO

Gym has been unmaintained since 2022 and does not support NumPy 2.0 amongst other critical functionality.
Please upgrade to Gymnasium, the maintained drop-in replacement of Gym, or contact the authors of your software and request that they upgrade.
Users of this version of Gym should be able to simply replace 'import gym' with 'import gymnasium as gym' in the vast majority of cases.
See the migration guide at https://gymnasium.farama.org/introduction/migration_guide/ for additional information.


In [6]:
agent_0 = "submission/submission_example2.py"
agent_1 = "submission/simple_agent.py"
env.run([agent_0, agent_1])
obs = env.steps[719][0].observation
# obs = env.steps[70][1].observation
print(obs)

{'remainingOverageTime': 60, 'step': 719, 'player': 0, 'farms': [{'money': 5957.0, 'tiles': [[{'kind': 'WEED'}, {'kind': 'WEED'}, None, None, None, 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED'], [None, None, None, None, None, 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED'], [None, None, None, None, None, 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED'], [None, None, None, None, None, 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED'], [None, {'kind': 'WEED'}, None, None, {'kind': 'PLANT', 'crop': 'MELON', 'planted_day': 24, 'watered_today': True, 'consecutive_unwatered': 0, 'yield_units': 1, 'max_lifespan_step': 888, 'fertilized_until_day': -1}, 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED'], ['LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED'], ['LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED'], ['LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOCKED', 'LOC

```python
{'kind': 'COOP', 'animal': 'GOOSE', 'placed_day': 0, 'yield_units': 0, 'consecutive_unfed': 1, 'fed_today': False, 'cared_today': False, 'fertilizer_available': True, 'pending_care_bonus': 0}

{'kind': 'PASTURE'} # have same structure as coop

{'kind': 'PLANT', 'crop': 'MELON', 'planted_day': 24, 'watered_today': True, 'consecutive_unwatered': 0, 'yield_units': 1, 'max_lifespan_step': 888, 'fertilized_until_day': -1}

{'kind': 'WEED'} # just weed area no data

'LOCKED'

None

['kind', 'animal'/'crop', 'yield_units', 'water_today'/ 'fed_today', 'fertilized_until_day'/'pending_care_bonus': 0]

In [ ]:
from kaggle_environments import make
from data_format import (
    preprocess,
)

agent_0 = "submission/submission_example2.py"
agent_1 = "submission/simple_agent.py"

env = make("kaggriculture", debug=False)

env.run([agent_0, agent_1])

# Get raw observation
obs = env.steps[719][0].observation

print("========== RAW OBSERVATION ==========")
print(obs)

# Encode observation
encoded_obs = preprocess(obs, player_index=0)

print("\n========== ENCODED OBSERVATION ==========")
print(encoded_obs)

print("\n========== SHAPE ==========")
print(encoded_obs.shape)

In [24]:
farms = float(obs.get("step", 0))
print(float(obs.get("day", 0)) *24)
print(float(obs.get("hour", 0)))
print(float(obs.get("remainingOverageTime", 0)))
farms

696.0
23.0
60.0


719.0